src/libraries/Microsoft.PowerFx.Connectors/ODataVisitor.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(RecordNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(ErrorNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(LazyEvalNode node, object context)
+        {
+            return node.Child.Accept(this, context);
+        }
+
+        public override string Visit(CallNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(BinaryOpNode node, object context)
+        {
+            return $"{node.Left.Accept(this, context)} {ODataOpFromBinaryOp(node.Op)} {node.Right.Accept(this, context)}";
+        }
+
+        public override string Visit(UnaryOpNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(ScopeAccessNode node, object context)
+        {
+            if (node.Value is ScopeAccessSymbol symbol)
+            {
+                return symbol.Name.Value;
+            }
+
+            return string.Empty;
+        }
+
+        public override string Visit(RecordFieldAccessNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(ResolvedObjectNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(SingleColumnTableAccessNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(ChainingNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override string Visit(AggregateCoercionNode node, object context)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        private static string ODataOpFromBinaryOp(BinaryOpKind op) =>
+            op switch
+            {
+                // BinaryOpKind.Invalid => expr,
+                // BinaryOpKind.InText => expr,
+                // BinaryOpKind.ExactInText => expr,
+                // BinaryOpKind.InScalarTable => expr,
+                // BinaryOpKind.ExactInScalarTable => expr,
+                // BinaryOpKind.InRecordTable => expr,
+                BinaryOpKind.AddNumbers => "+",
+                BinaryOpKind.AddDateAndTime => "+",
+                BinaryOpKind.AddDateAndDay => "+",
+                BinaryOpKind.AddDateTimeAndDay => "+",
+                BinaryOpKind.AddTimeAndMilliseconds => "+",
+                BinaryOpKind.DateDifference => "-",
+                BinaryOpKind.TimeDifference => "-",
```

Comment:
Some of these may be interesting, our date/time addition semantics may not be the same as what the target implements. We should check with Sagar/Reza how the JS runtime handles it.